Tests/SwiftFormatPrettyPrintTests/IfStmtTests.swift

```diff
@@ -283,4 +283,26 @@ public class IfStmtTests: PrettyPrintTestCase {
 
     assertPrettyPrintEqual(input: input, expected: expected, linelength: 50)
   }
+
+  public func testConditionExpressionOperatorGrouping() {
+    let input =
+      """
+      if someObj is SuperVerboselyNamedType || someObj is AnotherPrettyLongType  || someObjc == "APlainString" || someObj == 4 {
+        // do something
+      }
+      """
+
+    let expected =
+      """
+      if someObj is SuperVerboselyNamedType
+        || someObj is AnotherPrettyLongType
+        || someObjc == "APlainString" || someObj == 4
```

Comment:
I like the idea. I tend to dislike bin packing of conditional expressions in particular, because it's easier to miss expressions. The token between the `if` and the first condition is already a space, so I think this might be straightforward.